Extensions/Physics2Behavior/tests/physics2runtimebehavior.spec.js

```diff
@@ -0,0 +1,269 @@
+          // on the static object
+          assertCollision(movingObject, staticObject, {
+            started: false,
+            collision: false,
+            stopped: false,
+          });
+        } else {
+          hasBounced = true;
+          expect(movingObject.getY() < staticObject.getY()).to.be(true);
+          assertCollision(movingObject, staticObject, {
+            started: true,
+            collision: true,
+            stopped: true,
+          });
+        }
+
+        stepIndex++;
+      }
+      if (!hasBounced) {
+        throw new Error('Contact did not happen, nothing was tested.');
+      }
+    });
+
+    it('should begin to bounce during the frame (at high fps), collision should be detected, as well as start of the collision, but not the end', () => {
+      const fps = 50;
+      runtimeGame.setGameResolutionSize(1000, 1000);
+      runtimeScene._timeManager.getElapsedTime = function () {
+        return (1 / fps) * 1000;
+      };
+
+      const movingObject = createObject(runtimeScene);
+      const staticObject = createObject(runtimeScene, { bodyType: 'Static' });
+      staticObject.setPosition(0, 25);
+      movingObject.setPosition(0, 0);
+      const staticObjectBehavior = staticObject.getBehavior('Physics2');
+      const movingObjectBehavior = movingObject.getBehavior('Physics2');
+      if (!staticObjectBehavior || !movingObjectBehavior) {
+        throw new Error('Behaviors not found, test cannot be run.');
+      }
+      movingObjectBehavior.setLinearVelocityY(40000);
+
+      let hasBounced = false;
+      let stepIndex = 0;
+      while (stepIndex < 10 && !hasBounced) {
+        runtimeScene.renderAndStep(1000 / fps);
+        if (movingObjectBehavior.getLinearVelocityY() > 0) {
+          // If the moving object has a positive velocity, it hasn't bounced
+          // on the static object
+          assertCollision(movingObject, staticObject, {
+            started: false,
+            collision: false,
+            stopped: false,
+          });
+        } else {
+          hasBounced = true;
+          expect(movingObject.getY() < staticObject.getY()).to.be(true);
+          assertCollision(movingObject, staticObject, {
+            started: true,
+            collision: true,
+            stopped: false,
+          });
+        }
+
+        stepIndex++;
+      }
+      if (!hasBounced) {
+        throw new Error('Contact did not happen, nothing was tested.');
+      }
+
+      runtimeScene.renderAndStep(1000 / fps);
+
+      // At next frame, end of collision should be detected
+      assertCollision(movingObject, staticObject, {
+        started: false,
+        collision: false,
+        stopped: true,
+      });
+    });
+  });
+
+  describe('onContactBegin', () => {
+    let runtimeGame;
+    let runtimeScene;
+    beforeEach(() => {
+      [runtimeGame, runtimeScene] = createGameWithSceneWithPhysics2SharedData();
+    });
+
+    it('should add behavior to list of started contacts', () => {
```

Comment:
a description like:
"can detect a contact that start during a physic sub steps"
?

Maybe this test could do a "end" after the "beginning" to show that it's keep in this order?
Something like "detect a contact that is maintained only during some physics intermediate sub-steps" ?